[oneTBB] Add spec changes for std::invoke feature (#497)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/named_requirements/flow_graph/cpp17_multifunction_node_body.rst

.. SPDX-FileCopyrightText: 2019-2023 Intel Corporation
..
.. SPDX-License-Identifier: CC-BY-4.0

==========================
Cpp17MultifunctionNodeBody
==========================
**[req.cpp17_multifunction_node_body]**

> A type `Body` satisfies `Cpp17MultifunctionNodeBody` if it meets the following requirements:

Author: ```suggestion
The ``Body`` type satisfies `Cpp17MultifunctionNodeBody` if it meets the following requirements:
```

* The `CopyConstructible` requirements from [copyconstructible] ISO C++ Standard section

> * The `Destructible` requirements from [destructible] ISO C++ Standard section

Author: ```suggestion
* The `Destructible` requirements from the [destructible] section of the ISO C++ Standard.
```

> * The `Callable` requirements from [func.wrap.func] ISO C++ Standard section for arguments types ``const Input&, OutputPortsType&`` and return the type ``void``

Author: ```suggestion
* The `Callable` requirements from the [func.wrap.func] section of the ISO C++ Standard  for the ``const Input&, OutputPortsType&`` arguments types, and returns the ``void`` type.
```

>   where ``Input`` is the corresponding template argument of the ``multifunction_node``. The ``OutputPortsType`` is the same as the ``output_ports_type`` member type

Author: ```suggestion
Where the ``Input`` is the corresponding template argument of the ``multifunction_node``. The ``OutputPortsType`` is the same as the ``output_ports_type`` member type
```

  of the ``multifunction_node``.

> Considering ``body`` to be an object of type ``Body``, ``input`` to be an object of type ``const Input&`` and ``output_ports``` to be an object of type ``OutputPortsType&``,

Author: ```suggestion
Considering:

 * The ``body`` to be an object of the ``Body`` type
 * The ``input`` to be an object of the ``const Input&`` type
 * The ``output_ports``` to be an object of the ``OutputPortsType&`` type
```

applying ``std::invoke(body, input, output_ports)`` performs operation on ``input``. May call ``try_put()`` on zero or more of the output ports in ``output_ports``.
May call ``try_put()`` on any output port multiple times.

> **_NOTE:_**  It allows to pass pointer-to-function of the class ``Input`` as a body of ``multifunction_node``.

Author: ```suggestion
**_NOTE:_** You can pass a pointer-to-function of the ``Input`` class as a body of ``multifunction_node``.
```
